src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/BlobServiceClientProvider.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.blobrouter.clients.bulkscanprocessor.BulkScanProcessorClient;
+import uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount;
+
+@Service
+public class BlobServiceClientProvider {
+
+    private final BlobServiceClient crimeClient;
+    private final BulkScanProcessorClient bulkScanSasTokenClient;
+    private final String bulkScanStorageUrl;
+
+    public BlobServiceClientProvider(
+        @Qualifier("crime-storage-client") BlobServiceClient crimeClient,
+        BulkScanProcessorClient bulkScanSasTokenClient,
+        @Value("${storage.bulkscan.url}") String bulkScanStorageUrl
+    ) {
+        this.crimeClient = crimeClient;
+        this.bulkScanSasTokenClient = bulkScanSasTokenClient;
+        this.bulkScanStorageUrl = bulkScanStorageUrl;
+    }
+
+    public BlobServiceClient get(TargetStorageAccount targetStorageAccount, String containerName) {
+        switch (targetStorageAccount) {
+            case BULKSCAN:
+                // retrieving a SAS token every time we're getting a client, but this will be cached in the future
+                return new BlobServiceClientBuilder()
+                    .sasToken(bulkScanSasTokenClient.getSasToken(containerName).sasToken)
+                    .endpoint(bulkScanStorageUrl)
+                    .buildClient();
+            case CRIME:
+                return crimeClient;
+            default:
+                throw new UnknownStorageAccountException(
```

Comment:
this scenario is not covered by tests but should be fine to be done later as a separate Pr

Reply:
It's impossible to pass an unrecognised account value. That would only happen if someone added a new value to the enum but forgot to handle it in this class.